_posts/wasmer-go-embedding-1.0.md

```diff
@@ -0,0 +1,534 @@
+
+// … later… in another galaxy, far far away…
+
+serializedModule, _ := ioutil.ReadFile("my_wasm_module.so")
+module, _ := wasmer.DeserializeModule(store, serializedModule)
+
+// And instantiate it!
+instance, _ := wasmer.NewInstance(module, wasmer.NewImportObject())
+```
+
+Deserializing a compiled WebAssembly module with the Native engine is
+generally faster as it requires less operations.
+
+## Cross-compilation
+
+The engines and the compilers have a new very interesting feature:
+They can cross-compile. It means that from a machine A with a certain
+CPU, it is possible to compile for another machine B with a different
+CPU. Introducing the `Target`, `Triple` and `CpuFeatures` API.
+
+Let's say we are on a `aarch64-unknown-linux-gnu` machine, and we want
+to compile for an `x86_64-apple-darwin` machine:
+
+```go
+// Let's declare the triple of the machine we want to compile for.
+triple, _ := wasmer.NewTriple("x86_64-apple-darwin")
+
+// Let's configure the CPU features.
+cpuFeatures := wasmer.NewCpuFeatures()
+cpuFeatures.Add("sse2")
+
+// Finally, a Target is a pair composed of a Triple and CPU features.
+target := wasmer.NewTarget(triple, cpuFeatures)
+```
+
+After that, we pass the `Target` to the `Config` API, we compile the
+WebAssembly module, and save it:
+
+```go
+// Compile for a specific target.
+config := wasmer.NewConfig().UseTarget(target)
+
+// Create the engine with a specific configuration.
+engine := wasmer.NewEngineWithConfig(config)
+store := wasmer.NewStore(engine)
+
+// Let's compile the module for the given target.
+module, _ := wasmer.NewModule(store, wasmBytes)
+
+// Serialize the compiled module.
+ioutil.Writefile("my_wsam_module.wjit", module.Serialize(), 0644)
+```
+
+The serialized compiled module can be then deserialized on the
+targeted machine, instantiated, and executed!
+
+That way, it is easier to pre-compile any WebAssembly modules for a
+variety of machines!
+
+## WASI
+
+The `wasmer-go` package now features
+[WASI](https://github.com/WebAssembly/wasi) (The WebAssembly System
+Interface), with all the snapshot previews; understand, all the
+versions.
+
+First, the `GetWasiVersion` function can be used to know which version
+of WASI a WebAssembly module is using or whether it's not using WASI
+at all:
+
+```go
+module, _ := wasmer.NewModule(store, wasmBytes)
+
+fmt.Println(GetWasiVersion(module))
+```
+
+Second, to setup WASI, it's required to start by creating a
+`WasiEnvironment` with the help of the `NewWasiStateBuilder` API. The
+generated `WasiEnvironment` object is then used to generate an
+`ImportObject`. This later contains all the imports that “bridge” the
+WebAssembly module to the host to make WASI a reality. Of course, it's
+possible to use this `ImportObject` to import your own host functions,
+memories etc., just like with any `ImportObject`. `WasiEnvironment` is
+also responsible to read the `stdout` and `stderr` streams if they are
+captured.
```

Comment:
```suggestion
Second, to setup WASI, we start by creating a
`WasiEnvironment` with the help of the `NewWasiStateBuilder` API. We then use the
generated `WasiEnvironment` object to generate an
`ImportObject`. This contains all the imports that “bridge” the
WebAssembly module to the host to make WASI a reality. Of course, it's
possible to use this `ImportObject` to import your own host functions,
memories etc., just like with any other `ImportObject`. `WasiEnvironment` is
also responsible for redirecting the `stdout` and `stderr` streams if they are
captured.
```

Feel free to change what I said in regards to `redirecting`. depending on how it actually works, that sounded more clear to me though

Reply:
It does sound clearer. Thanks!